firestore-bigquery-export/firestore-bigquery-change-tracker/src/bigquery/index.ts

```diff
@@ -51,29 +51,33 @@ export class FirestoreBigQueryEventHistoryTracker
 
   async record(events: FirestoreDocumentChangeEvent[]) {
     await this.initialize();
-
+    const options = {
+      raw: true
+    };
     const rows = events.map((event) => {
-      // This must match firestoreToBQTable().
       return {
-        timestamp: event.timestamp,
-        event_id: event.eventId,
-        document_name: event.documentName,
-        operation: ChangeType[event.operation],
-        data: JSON.stringify(event.data),
+        insertId: event.eventId,
+        json: {
```

Comment:
Will nesting this data not create a breaking change for users with existing data?

Reply:
This doesn't change the content of the row that is mirrored in bigquery, it is just the required format for the streaming insert payload if we want to use insertIds (https://cloud.google.com/bigquery/docs/reference/rest/v2/tabledata/insertAll). The format of the json payload changes so that we can specify a custom insertId, but that id is transparent to existing users.

Reply:
Makes sense thanks for explaining